templates/taxi/car_detail.html

```diff
@@ -3,20 +3,33 @@
 {% block content %}
   <h1>
     {{ car.model }}
-    <a href="{% url 'taxi:car-delete' pk=car.id %}" class="btn btn-danger link-to-page">
-      Delete
-    </a>
-
-    <a href="{% url 'taxi:car-update' pk=car.id %}" class="btn btn-secondary link-to-page">
-      Update
-    </a>
   </h1>
   <p>Manufacturer: ({{ car.manufacturer.name }}, {{ car.manufacturer.country }})</p>
-  <h1>Drivers</h1>
-  <hr>
+  <h1>Drivers
+    <form method="post" action="">
+      {% csrf_token %}
+      <button type="submit"
+          {% if request.user in car.drivers.all %} class="btn btn-danger" {% else %}
+              class="btn btn-success" {% endif %}>
+        {% if request.user in car.drivers.all %}
+          Delete me from this car
+        {% else %}
+          Assign me to this car
+        {% endif %}
+      </button>
+    </form>
+  </h1>
+    <hr>
   <ul>
     {% for driver in car.drivers.all %}
       <li>{{ driver.username }} ({{ driver.first_name }} {{ driver.last_name }})</li>
     {% endfor %}
   </ul>
+  <a style="float: left" href="{% url 'taxi:car-delete' pk=car.id %}" class="btn btn-danger link-to-page">
+    Delete
+  </a>
```

Comment:
Consider adding a confirmation dialog or step before executing the delete action to prevent accidental deletions. This can improve user experience by ensuring that users are certain about their decision to delete.